### PR TITLE
Adding Brave Wallet buildflag, menu item

### PR DESCRIPTION
**Pull request by ryanml**
Resolves: https://github.com/brave/brave-browser/issues/4578
Resolves: https://github.com/brave/brave-browser/issues/4579

Screen Shot 2019-05-24 at 12 15 03 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
1. Confirm that the menu item is not shown in official builds
2. Confirm that clicking on the menu item opens a new tab to `chrome://wallet`
3. Do a test of the other menu items to ensure functionality is preserved.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/toolbar/brave_app_menu_model.cc [hunk 1]
```diff
@@ -35,9 +35,13 @@ void BraveAppMenuModel::InsertBraveMenuItems() {
         GetIndexOfCommandId(IDC_SHOW_DOWNLOADS),
         IDC_SHOW_BRAVE_REWARDS,
         IDS_SHOW_BRAVE_REWARDS);
+    InsertItemWithStringIdAt(
```
Comment (NejcZdovc) [inline browser/ui/toolbar/brave_app_menu_model.cc]: This one should be probably protected by build flag check

Comment (ryanml) [reply browser/ui/toolbar/brave_app_menu_model.cc]: @NejcZdovc this check here should be sufficient: https://github.com/brave/brave-core/pull/2496/files#diff-48d8778906af24da93d2c7079c17ec40R89

Are you seeing a case where the above does not work properly?

File: common/webui_url_constants.cc [hunk 2]
```diff
@@ -21,3 +21,4 @@ const char kBraveUIRewardsURL[] = "chrome://rewards/";
 const char kBraveUIAdblockURL[] = "chrome://adblock/";
 const char kBraveUITipURL[] = "chrome://tip/";
 const char kBraveUISyncURL[] = "chrome://sync/";
+const char kBraveUIWalletURL[] = "chrome://wallet";
```
Comment (bbondy) [inline common/webui_url_constants.cc]: Please add trailing slash: `const char kBraveUIWalletURL[] = "chrome://wallet/";`

Comment (ryanml) [reply common/webui_url_constants.cc]: done

**Pull request merged**